servers/annotations-api/src/dataservices/highlights.ts

```diff
@@ -225,36 +246,57 @@ export class HighlightsDataService {
     if (!row) {
       throw new NotFoundError('No annotation found for the given ID');
     }
-
     return row;
   }
 
   /**
    * Delete highlight for a given ID
    * @param highlightId
+   * @param trx Optionally, a transaction object.
+   * Can be used to chain multiple operations.
+   * If one is not given, a new transaction will be opened.
    */
-  public async delete(highlightId: string): Promise<string> {
+  public async delete(
+    highlightId: string,
+    trx?: Knex.Transaction,
+  ): Promise<string> {
     const annotation = await this.getHighlightByIdQuery(highlightId);
+    const transaction = trx ?? (await this.writeDb.transaction());
+    try {
+      const result = await this._delete(highlightId, annotation, transaction);
+      transaction.commit();
+      return result;
+    } catch (error) {
+      // Just in case, roll back if not already done
+      if (!transaction.isCompleted()) {
+        transaction.rollback();
+      }
+      throw error;
+    }
+  }
 
-    return await this.writeDb.transaction(async (trx: Knex.Transaction) => {
-      // This will throw and error if it doesn't like you
-      await this.writeDb<HighlightEntity>('user_annotations')
-        .update({ status: 0 })
-        .where('user_id', this.userId)
-        .andWhere('annotation_id', highlightId);
+  private async _delete(
+    highlightId: string,
+    annotation: HighlightEntity,
+    trx: Knex.Transaction,
+  ) {
+    // This will throw and error if it doesn't like you
```

Comment:
🤣

Reply:
heh, I didn't write that line but I did preserve it